src/persist-client/src/stats.rs

```diff
@@ -121,16 +121,31 @@ pub(crate) fn untrimmable_columns(cfg: &ConfigSet) -> UntrimmableColumns {
     }
 }
 
-pub(crate) fn part_stats_for_legacy_part<K: Codec, V: Codec>(
+/// Encodes a [`ColumnarRecords`] into a [`Part`] and calculates [`PartStats`].
+///
+/// Note: The [`Part`] will contain the same data as [`ColumnarRecords`], but
+/// the [`Part`] will have the data fully structured as opposed to an opaque
+/// binary blob.
+pub(crate) fn encode_updates<K, V>(
     schemas: &Schemas<K, V>,
-    part: &[ColumnarRecords],
-) -> Result<PartStats, String> {
-    // This is a laughably inefficient placeholder implementation of stats
-    // on the old part format. We don't intend to make this fast, rather we
-    // intend to compute stats on the new part format.
+    updates: &BlobTraceUpdates,
+) -> Result<(Part, PartStats), String>
+where
+    K: Codec,
+    V: Codec,
+{
+    let updates = match updates {
+        BlobTraceUpdates::Row(updates) => itertools::Either::Left(updates.into_iter()),
+        BlobTraceUpdates::Both((codec, _structured)) => {
+            // This is super unexpected, but not worthy of a panic.
+            tracing::error!("re-encoding structured data?");
```

Comment:
should we do something here to make this fail in CI?

Reply:
done